lib/server/index.js

```diff
@@ -223,6 +227,11 @@ Server.prototype = {
     var config = this.config;
     var test_page = config.get('test_page');
 
+    // If there is an array of test pages, we cycle through them as the default
+    if (Array.isArray(test_page)) {
+      test_page = test_page[this.testPageCount % test_page.length];
+    }
```

Comment:
This approach breaks when you would have  2 browsers, 2 pages and parallel set to 4 as you can't depend on the order of browsers connecting and may end up browser A testing page 1 twice and browser B page 2.

I see two solutions:
1) directly set the test page as launch url per runner
2) store in a map the unique id and the targeted test page at creation time and forward based on that